blocks/bricks/conv.py

```diff
@@ -221,7 +273,83 @@ def get_dim(self, name):
         return super(ConvolutionalLayer, self).get_dim(name)
 
 
+class ConvolutionalNetwork(Sequence, Initializable, Feedforward):
+    """A convolutional network.
+
+    Parameters
+    ----------
+    layers : list
+        List of convolutional layers
+        (e.g. ConvolutionalActivation or ConvolutionalLayer)
+    batch_size : int
+        Number of images in batch.
+    num_channels : int
+        Number of input channels in the image. For the first layer this is
+        normally 1 for grayscale images and 3 for color (RGB) images. For
+        subsequent layers this is equal to the number of filters output by
+        the previous convolutional layer. The filters are pooled over the
+        channels.
+    input_dim : tuple
+        Width and height of the input (image/featuremap).
+
+    Notes
+    -----
+    The passed convolutional layers should be 'lazy' constructed, that is,
+    without specifying the batch_size, num_channels and input_dim.
+    ConvolutionalNetwork will set (before allocation) the input dimensions
+    of a layer to the output dimensions of the previous layer
+    by the _push_allocation_config method.
+
+    Example
+    -------
+    TODO
+    """
+
+    def __init__(self, layers, batch_size, num_channels, input_dim, **kwargs):
+        self.layers = layers
+        self.input_dim = input_dim
+        self.num_channels = num_channels
+        self.batch_size = batch_size
+
+        application_methods = [brick.apply for brick in layers]
+        super(ConvolutionalNetwork, self).__init__(
+            application_methods=application_methods, **kwargs)
+
+    def get_dim(self, name):
+        if name == 'input_':
+            return self.input_dim
+        if name == 'output':
+            return self.layers[-1].get_dim(name)
+        return super(ConvolutionalNetwork, self).get_dim(name)
+
+    def _push_allocation_config(self):
+        input_dim = self.input_dim
+        num_channels = self.num_channels
+        for layer in self.layers:
+            layer.input_dim = input_dim
+            layer.num_channels = num_channels
+            layer.batch_size = self.batch_size
+
+            # Push input dimensions to children
+            layer._push_allocation_config()
```

Comment:
Got it, my bad. Normally you don't need to push this manually, because the `push_allocation_config` method automatically does this for you. In this case however you want to do it manually because you want to access `layer.get_dim('output')` directly afterwards to get `output_shape`.